components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/jwt_auth_provider.bal

```diff
@@ -40,20 +41,23 @@ public type JwtAuthProvider object {
     public string className;
     public boolean classLoaded;
     public APIGatewayCache gatewayCache = new;
+    public boolean remoteUserClaimRetrievalEnabled = false;
 
     # Provides authentication based on the provided JWT token.
     #
     # + jwtValidatorConfig - JWT validator configurations
     # + subscriptionValEnabled - Validate subscription
+    # + remoteUserClaimRetrievalEnabled - true if the remote user claim retrieval is required
     public function __init(jwt:JwtValidatorConfig jwtValidatorConfig, boolean subscriptionValEnabled, string consumerKeyClaim,
-        map<anydata>[] | error claims, string className, boolean classLoaded) {
+        map<anydata>[] | error claims, string className, boolean classLoaded, boolean remoteUserClaimRetrievalEnabled) {
         self.jwtValidatorConfig = jwtValidatorConfig;
         self.inboundJwtAuthProvider = new (jwtValidatorConfig);
         self.subscriptionValEnabled = subscriptionValEnabled;
         self.consumerKeyClaim = consumerKeyClaim;
         self.claims = claims;
         self.className = className;
         self.classLoaded = classLoaded;
+        self.remoteUserClaimRetrievalEnabled = remoteUserClaimRetrievalEnabled;
```

Comment:
Do we need this config at provider level. Can't we have it only in handler level

Reply:
Added in this level because need this for each issuer separately. Don't we need such granularity?